task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -319,14 +319,17 @@ class WatchedSubprocess:
     @classmethod
     def start(
         cls,
-        path: str | os.PathLike[str],
-        what: TaskInstance,
-        client: Client,
+        id: UUID | None,
```

Comment:
Same, shouldn't be none

Reply:
but it was for one of the subclasses

Reply:
this is just making explicit and correct what was already the case

Reply:
oh wait it maybe wasn't lemme see

Reply:
ok so, here's what's going on here.
previously `id` was not an explicit param.  it was part of "constructor kwargs" and it was sometimes provided and sometimes no.  it was provided by dag processor but _not_ for task execution.  for task execution, it was grabbed from `ti.id` _inside_ this function.  here, i add it as an explicit argument and it is thus always supplied. and we can make it not None now.

Reply:
resolving this one